lighthouse-core/lib/asset-saver.js

```diff
@@ -194,12 +194,14 @@ async function prepareAssets(artifacts, audits) {
 }
 
 /**
- * Generates a JSON representation of traceData line-by-line to avoid OOM due to
- * very large traces.
+ * Generates a JSON representation of traceData line-by-line to avoid OOM due to very large traces.
+ * COMPAT: As of Node 9, JSON.parse/stringify can handle 256MB+ strings. Once we drop support for
+ * Node 8, we can 'revert' PR #2593. See https://stackoverflow.com/a/47781288/89484
```

Comment:
> Once we drop support for Node 8, we can 'revert' PR #2593

not sure if we actually want to anymore...we like the event-per-line formatting, right?

(We could consider shipping the `pretty-json-stringify` we just used in the snyk snapshot thing—no dependencies and 2.43KB before minification!—but it looks like using it on large traces will [depend on Node 9 as well](https://github.com/sergets/pretty-json-stringify/blob/5e81f3108c840d7ec7e967b02a0e4b6b17b98726/index.js#L20))